### PR TITLE
#F01 Footer Content & Styling #53

### PR DESCRIPTION
**Pull request by JimLynx**
- Created footer.html layout & Styled with CSS
- Created templates html page links
- Created basic views and paths for all links in footer

Views/Paths/templates created for:

1. Privacy Policy
2. Plagiarism Policy
3. FAQ
4. Judging Criteria
5. Useful Resources

Testing:

- Passed CSS through CSS Validation and no issues present
- Passed HTML through HTML Validator and no issues present
- Clicked on all links to ensure they open in a new tab and presented as expected

IMAGES:

1. Footer
Footer

2. Link template example
template

### CODE REVIEW

File: static/css/style.css [hunk 1]
```diff
@@ -1,74 +1,130 @@
...
-    --s-teal: #23BBBB;
-    --t-grey: #E0E7FF;
-    --white: #FFFFFF;
-    --black: #000000;
-    --bg-grey: #F5F5F5;
-}
-
-html, body {
-    height: 100%;
+  /* P = Primary | S = Secondary | T = Tertiary */
```
Comment (TravelTimN) [inline static/css/style.css:2]: @JimLynx as discussed, can you investigate this (probably an extension you have), and revert these back please?

File: static/css/style.css [hunk 2]
```diff
@@ -1,74 +1,130 @@
...
 /* BUTTONS */
 .btn-ci {
-    background-color: transparent;
-    color: var(--p-orange);
-    border: 3px solid var(--p-orange);
-    transition: all .4s;
+  background-color: transparent;
+  color: var(--p-orange);
+  border: 3px solid var(--p-orange);
+  transition: all 0.4s;
```
Comment (TravelTimN) [inline static/css/style.css:62]: @JimLynx 
Probably from your extensions again, but our style guide (apologies, you wouldn't have known this), don't include the leading 0s for these)

File: static/css/style.css [hunk 3]
```diff
@@ -1,74 +1,130 @@
...
+}
+
+footer p {
+  text-align: justify;
+}
+
+footer a:hover {
+  color: var(--p-orange);
+  text-decoration: none;
+}
```
Comment (TravelTimN) [inline static/css/style.css:90]: @JimLynx 
A better method perhaps, is to do this globally.
Set text-decoration on all link hovers to be none, not specific to just the footer links.

File: static/css/style.css [hunk 4]
```diff
@@ -1,74 +1,130 @@
...
+
+.copyright-wrapper a {
+  color: var(--s-grey);
+}
+
+.fab {
+  font-size: 200%;
+  margin: 0 20px;
+  color: var(--s-grey);
+  transition: transform 0.4s;
```
Comment (TravelTimN) [inline static/css/style.css:124]: @JimLynx 
Here as well, `.4s` instead of `0.4s`. Thanks

File: templates/faq.html [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+<p> Lorem ipsum dolor sit amet consectetur, adipisicing elit. Beatae, error cumque. Esse, obcaecati necessitatibus quasi
+    doloremque soluta cum voluptatum veritatis eius voluptas molestias a, magnam quam corporis. Voluptas, placeat
+    beatae.
+</p>
+<p>
+    Lorem ipsum dolor, sit amet consectetur adipisicing elit. Magnam fugiat fugit hic debitis eligendi unde eaque
+    facilis dignissimos commodi eos, odit nesciunt. Rerum quos suscipit quibusdam dolorum impedit temporibus sunt!
+</p>
+
+{% endblock %}
```
Comment (TravelTimN) [inline templates/faq.html:18]: @JimLynx 
You should always have a blank line (one only) at the end of each file.

File: templates/faq.html [hunk 6]
```diff
@@ -0,0 +1,18 @@
+{% extends 'base.html' %}
+{% load static %}
+{% block content %}
+
+<!-- help needed -->
+<!-- needs to be handled from backend urls and views -->
+<h1>FAQ</h1>
+
+<p> Lorem ipsum dolor sit amet consectetur, adipisicing elit. Beatae, error cumque. Esse, obcaecati necessitatibus quasi
```
Comment (TravelTimN) [inline templates/faq.html:9]: @JimLynx 
I think better here, would be to just leave out the Lorem text and paragraphs, and let's focus on getting that perhaps added as a new User Story #Mxx.

File: templates/faq.html [hunk 7]
```diff
@@ -0,0 +1,18 @@
+{% extends 'base.html' %}
+{% load static %}
+{% block content %}
+
+<!-- help needed -->
+<!-- needs to be handled from backend urls and views -->
```
Comment (TravelTimN) [inline templates/faq.html:6]: @JimLynx 
Looks like you've coded in the backend urls/views now? Can this be removed?

File: templates/includes/footer.html [hunk 8]
```diff
@@ -1,2 +1,79 @@
-<!-- TO DO - help wanted -->
-<h3>TO DO : templates > includes > footer.html</h3>
+<div class="container">
+    <div class="row">
+        <div class="col-sm-6 col-md-3">
+            <h4>Policies</h4>
+            <ul>
+                <li>
+                    <!-- To link to own Privacy Policy page once built -->
```
Comment (TravelTimN) [inline templates/includes/footer.html:7]: @JimLynx
You've built the views/urls now, so can this comment be removed?

File: templates/includes/footer.html [hunk 9]
```diff
@@ -1,2 +1,79 @@
...
+    <div class="row">
+        <div class="col-sm-6 col-md-3">
+            <h4>Policies</h4>
+            <ul>
+                <li>
+                    <!-- To link to own Privacy Policy page once built -->
+                    <a href="{% url 'privacy_policy' %}" target=" _blank">Privacy Policy</a>
+                </li>
+                <li>
+                    <!-- To link to own Plagiarism Policy page once built -->
```
Comment (TravelTimN) [inline templates/includes/footer.html:11]: @JimLynx 
Same here - comment is no longer needed?

File: templates/includes/footer.html [hunk 10]
```diff
@@ -1,2 +1,79 @@
...
+                    <a href="{% url 'plagiarism_policy' %}" target="_blank">Plagiarism Policy</a>
+                </li>
+                <li>
+                    <a href="https://docs.github.com/en/free-pro-team@latest/github/site-policy/github-terms-of-service"
+                        target="_blank">Github Terms of Service</a>
```
Comment (TravelTimN) [inline templates/includes/footer.html:16]: @JimLynx 
GitHub* not Github

File: templates/includes/footer.html [hunk 11]
```diff
@@ -1,2 +1,79 @@
...
+                </li>
+            </ul>
+        </div>
+        <div class="col-sm-6 col-md-3">
+            <h4>
+                Resources
+            </h4>
+            <ul>
+                <li>
+                    <!-- to link to FAQ page once built -->
```
Comment (TravelTimN) [inline templates/includes/footer.html:26]: @JimLynx 
comment obsolete?

File: templates/includes/footer.html [hunk 12]
```diff
@@ -1,2 +1,79 @@
...
+                    <a href="{% url 'faq' %}" target="_blank">FAQ</a>
+                </li>
+                <li>
+                    <!-- to link to Judging Criteria page once built -->
```
Comment (TravelTimN) [inline templates/includes/footer.html:30]: @JimLynx 
comment obsolete?

File: templates/includes/footer.html [hunk 13]
```diff
@@ -1,2 +1,79 @@
...
+                    <a href="{% url 'judging_criteria' %}" target="_blank">Judging Criteria</a>
+                </li>
+                <li>
+                    <!-- to link to Useful Resources page once built -->
```
Comment (TravelTimN) [inline templates/includes/footer.html:34]: @JimLynx 
comment obsolete?

File: templates/includes/footer.html [hunk 14]
```diff
@@ -1,2 +1,79 @@
...
+                <i class="fab fa-linkedin"></i>
+            </a>
+            <a href="https://github.com/Code-Institute-Community/ci-hackathon-app" target="_blank">
+                <i class="fab fa-github"></i>
+            </a>
+        </div>
+    </div>
+    <div class="col copyright-wrapper">
+        <p>
+            <a href="https://codeinstitute.net/" target=" _blank">&copy; Copyright 2020 Code Institute</a>
```
Comment (TravelTimN) [inline templates/includes/footer.html:75]: @JimLynx 
`&copy;` with **Copyright** is redundant, let's remove the word 'Copyright'.

File: templates/includes/footer.html [hunk 15]
```diff
@@ -1,2 +1,79 @@
...
+        </p>
+    </div>
+
+</div>
```
Comment (TravelTimN) [inline templates/includes/footer.html:79]: @JimLynx 
Needs a blank line at the end of the file.

File: templates/judging-criteria.html [hunk 16]
```diff
@@ -0,0 +1,18 @@
+{% extends 'base.html' %}
```
Comment (TravelTimN) [inline templates/judging-criteria.html:1]: @JimLynx 
"base.html" double-quotes to keep it consistent with the original files.

File: templates/judging-criteria.html [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+<p> Lorem ipsum dolor sit amet consectetur, adipisicing elit. Beatae, error cumque. Esse, obcaecati necessitatibus quasi
+    doloremque soluta cum voluptatum veritatis eius voluptas molestias a, magnam quam corporis. Voluptas, placeat
+    beatae.
+</p>
+<p>
+    Lorem ipsum dolor, sit amet consectetur adipisicing elit. Magnam fugiat fugit hic debitis eligendi unde eaque
+    facilis dignissimos commodi eos, odit nesciunt. Rerum quos suscipit quibusdam dolorum impedit temporibus sunt!
+</p>
+
+{% endblock %}
```
Comment (TravelTimN) [inline templates/judging-criteria.html:18]: @JimLynx 
Needs a blank line at the end of the file

File: templates/judging-criteria.html [hunk 18]
```diff
@@ -0,0 +1,18 @@
+{% extends 'base.html' %}
+{% load static %}
+{% block content %}
+
+<!-- help needed -->
+<!-- needs to be handled from backend urls and views -->
```
Comment (TravelTimN) [inline templates/judging-criteria.html:6]: @JimLynx 
Is this obsolete now?

File: templates/plagiarism-policy.html [hunk 19]
```diff
@@ -0,0 +1,18 @@
+{% extends 'base.html' %}
```
Comment (TravelTimN) [inline templates/plagiarism-policy.html:1]: @JimLynx 
"base.html" double-quotes to keep it consistent with the original files.

File: templates/faq.html [hunk 20]
```diff
@@ -0,0 +1,18 @@
+{% extends 'base.html' %}
```
Comment (TravelTimN) [inline templates/faq.html:1]: @JimLynx 
"base.html" double-quotes to keep it consistent with the original files.

File: templates/plagiarism-policy.html [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+{% load static %}
+{% block content %}
+
+<!-- help needed -->
+<!-- needs to be handled from backend urls and views -->
```
Comment (TravelTimN) [inline templates/plagiarism-policy.html:6]: @JimLynx 
Is this comment obsolete?

File: templates/plagiarism-policy.html [hunk 22]
```diff
@@ -0,0 +1,18 @@
...
+<p> Lorem ipsum dolor sit amet consectetur, adipisicing elit. Beatae, error cumque. Esse, obcaecati necessitatibus quasi
+    doloremque soluta cum voluptatum veritatis eius voluptas molestias a, magnam quam corporis. Voluptas, placeat
+    beatae.
+</p>
+<p>
+    Lorem ipsum dolor, sit amet consectetur adipisicing elit. Magnam fugiat fugit hic debitis eligendi unde eaque
+    facilis dignissimos commodi eos, odit nesciunt. Rerum quos suscipit quibusdam dolorum impedit temporibus sunt!
+</p>
+
+{% endblock %}
```
Comment (TravelTimN) [inline templates/plagiarism-policy.html:18]: @JimLynx 
needs a new line at EOF

File: templates/privacy-policy.html [hunk 23]
```diff
@@ -0,0 +1,18 @@
+{% extends 'base.html' %}
```
Comment (TravelTimN) [inline templates/privacy-policy.html:1]: @JimLynx 
"base.html" double-quotes to keep it consistent with the original files.

File: templates/privacy-policy.html [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+{% load static %}
+{% block content %}
+
+<!-- help needed -->
+<!-- needs to be handled from backend urls and views -->
```
Comment (TravelTimN) [inline templates/privacy-policy.html:6]: @JimLynx 
obsolete comment

File: templates/privacy-policy.html [hunk 25]
```diff
@@ -0,0 +1,18 @@
...
+<p> Lorem ipsum dolor sit amet consectetur, adipisicing elit. Beatae, error cumque. Esse, obcaecati necessitatibus quasi
+    doloremque soluta cum voluptatum veritatis eius voluptas molestias a, magnam quam corporis. Voluptas, placeat
+    beatae.
+</p>
+<p>
+    Lorem ipsum dolor, sit amet consectetur adipisicing elit. Magnam fugiat fugit hic debitis eligendi unde eaque
+    facilis dignissimos commodi eos, odit nesciunt. Rerum quos suscipit quibusdam dolorum impedit temporibus sunt!
+</p>
+
+{% endblock %}
```
Comment (TravelTimN) [inline templates/privacy-policy.html:18]: @JimLynx 
new line at EOF

File: templates/useful-resources.html [hunk 26]
```diff
@@ -0,0 +1,18 @@
+{% extends 'base.html' %}
```
Comment (TravelTimN) [inline templates/useful-resources.html:1]: @JimLynx 
"base.html" double-quotes to keep it consistent with the original files.

File: templates/useful-resources.html [hunk 27]
```diff
@@ -0,0 +1,18 @@
...
+{% load static %}
+{% block content %}
+
+<!-- help needed -->
+<!-- needs to be handled from backend urls and views -->
```
Comment (TravelTimN) [inline templates/useful-resources.html:6]: @JimLynx 
obsolete

File: templates/useful-resources.html [hunk 28]
```diff
@@ -0,0 +1,18 @@
...
+<p> Lorem ipsum dolor sit amet consectetur, adipisicing elit. Beatae, error cumque. Esse, obcaecati necessitatibus quasi
+    doloremque soluta cum voluptatum veritatis eius voluptas molestias a, magnam quam corporis. Voluptas, placeat
+    beatae.
+</p>
+<p>
+    Lorem ipsum dolor, sit amet consectetur adipisicing elit. Magnam fugiat fugit hic debitis eligendi unde eaque
+    facilis dignissimos commodi eos, odit nesciunt. Rerum quos suscipit quibusdam dolorum impedit temporibus sunt!
+</p>
+
+{% endblock %}
```
Comment (TravelTimN) [inline templates/useful-resources.html:18]: @JimLynx 
new line at EOF

**Review by TravelTimN — CHANGES_REQUESTED** [review]
@JimLynx 
Looks great - there are some changes to be adjusted below.
Looks like you have a Merge Conflict to be updated as well, with the .gitignore file.
Not mentioned below on the Review, are the spaces on your CSS file.
Make sure you keep it consistent, which I believe is 4-spaces, whereas your extension looks to be adding 3-spaces only.

File: home/urls.py [hunk 29]
```diff
@@ -2,5 +2,10 @@
 from . import views
 
 urlpatterns = [
-    path("", views.index, name="home")
+    path("", views.index, name="home"),
+    path("faq/", views.faq, name="faq"),
+    path("judging_criteria/", views.judging_criteria, name="judging_criteria"),
+    path("plagiarism_policy/", views.plagiarism_policy, name="plagiarism_policy"),
+    path("privacy_policy/", views.privacy_policy, name="privacy_policy"),
+    path("useful_resources/", views.useful_resources, name="useful_resources")
```
Comment (TravelTimN) [inline home/urls.py:10]: @JimLynx 
Make sure to include a `comma` at the end of each line, so after useful_resources `path()`.

File: static/css/style.css (hunk 2 above)
Comment (JimLynx) [reply static/css/style.css:62]: Noted, this must have been the extension again as I didn't edit any CSS above the footer section. Will resolve shortly. :)

**Comment by JimLynx** [comment]
All changed in review completed and pushed updated files to **footer-content** branch

**Comment by stefdworschak** [comment]
@JimLynx could you just merge the newest changes from master and resolve the merge conflict please?

**Comment by JimLynx** [comment]
All good @stefdworschak ?

**Comment by TravelTimN** [comment]
The CSS file looks to be entirely modified again - something with an extension that you might want to be mindful of, @JimLynx ... other than that, let's see what @stefdworschak has to say.

**Review by stefdworschak — APPROVED** [review]
LGTM

@JimLynx as @TravelTimN mentioned I would check your extensions again, but I don't really see a difference between the two versions. So it should be fine.

**Pull request merged**